src/frontend/pages/Signup/Signup.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import "../Login/loginpage.css";
+import { Link } from "react-router-dom";
+import { useAuth } from "../../contexts/auth-context";
+
+export function Signup() {
+  const { authState, authDispatch, signup } = useAuth();
+
+  return (
+    <div className="flex">
+      <div className="login-card">
+        <h2 className="text-center">Signup</h2>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            signup();
+          }}
+        >
+          <div className="pt-1">
+            <label htmlFor="email">Email</label>
+            <br />
+            <input
+              type="email"
+              name="email"
+              id="email"
+              placeholder="abcd@gmail.com"
+              className="input"
+              value={authState.email}
+              required
```

Comment:
- learned how to do the auth part
- easy to understand logic 
- clean code and properly formatted ✅